linodecli/plugins/metadata.py

```
@@ -0,0 +1,203 @@
+    """
+    data = client.get_user_data()
+    rprint(data)
+
+
+def get_network(client: MetadataClient):
+    """
+    Get information about your instance’s IP addresses
+    """
+    data = client.get_network()
+    print_networking_tables(data)
+
+
+def get_ssh_keys(client: MetadataClient):
+    """
+    Get information about public SSH Keys configured on your instance
+    """
+    data = client.get_ssh_keys()
+    print_ssh_keys_table(data)
+
+
+COMMAND_MAP = {
+    "instance": get_instance,
+    "user-data": get_user_data,
+    "networking": get_network,
+    "sshkeys": get_ssh_keys,
+}
+
+
+def print_help(parser: argparse.ArgumentParser):
+    """
+    Print out the help info to the standard output.
+    """
+    parser.print_help()
+
+    # additional help
+    print()
+    print("Available endpoints: ")
+
+    command_help_map = [
+        [name, func.__doc__.strip()]
+        for name, func in sorted(COMMAND_MAP.items())
+    ]
+
+    tab = Table(show_header=False)
+    for row in command_help_map:
+        tab.add_row(*row)
+    rprint(tab)
+
+
+def get_metadata_parser():
+    parser = argparse.ArgumentParser(PLUGIN_BASE, add_help=False)
+
+    parser.add_argument(
+        "endpoint",
+        metavar="ENDPOINT",
+        nargs="?",
+        type=str,
+        help="The API endpoint to be called from the Metadata service.",
+    )
+
+    return parser
+
+
+def call(args, context):
+    """
+    The entrypoint for this plugin
+    """
+
+    parser = get_metadata_parser()
+    parsed, args = parser.parse_known_args(args)
+
+    if not parsed.endpoint in COMMAND_MAP or len(args) != 0:
+        print_help(parser)
+        sys.exit(0)
+
+    # make a client, but only if we weren't printing help and endpoint is valid
+    if not "--help" in args:
+        try:
+            client = MetadataClient()
+        except ConnectTimeout:
+            print(
+                "Can't access Metadata service. Please verify that you are inside a Linode."
+            )
```

Given this is an error, it might make sense to print this to stderr:
```suggestion
            print(
                "Can't access Metadata service. Please verify that you are inside a Linode.",
                file=sys.stderr
            )
```

Alternatively, you could raise an error here (e.g. `raise ConnectionError("Can't access Metadata service. Please verify that you are inside a Linode."`)